### ISSUE TITLE
MXNet on Windows via MinGW

### ISSUE THREAD

**Issue by sergeykolychev** [apache/incubator-mxnet#11835]
Hi guys,
Are there any plans to get Windows to compile via MinGW compiler on Windows ?
Tried, and as many before me, failed. Looks like there's a need for a substantial code modifications to get it compiled via gcc on Windows.
@szha

**Comment by chinakook** [comment]
Does MinGW has CUDA support?

**Comment by sergeykolychev** [comment]
@chinakook 
No, it does not at this time (nvidia only supports gcc on Linux).
However we do support compiling mxnet on MacOS (which usually does not have CUDA enabled devices) via gcc.
The reason I asked is that current Perl distributions all have Perl on Windows compiled via https://mingw-w64.org/doku.php
If MXNet is able to compile via MinGW 64 on Windows then I should be able to have Perl API to work on Windows as well. For now only for CPU, in future hopefully for AMD GPUs and eventually CUDA (that depends on Nvidia).

**Comment by chinakook** [comment]
Aha
